### PR TITLE
remove nuget-task-common

### PR DESCRIPTION
**Pull request by jotaylo**
Removes nuget-task-common, and merges shared nuget bits into packaging-common

tasks impacted:

### CODE REVIEW

File: make-options.json [hunk 1]
```diff
@@ -1,126 +1,16 @@
...
-        "CopyFilesOverSSHV0",
-        "CUrlUploaderV2",
-        "DecryptFileV1",
-        "DelayV1",
-        "DeleteFilesV1",
-        "DeployVisualStudioTestAgentV2",
-        "DockerV0",
-        "DockerV1",
-        "DockerComposeV0",
+        "NpmAuthenticateV0",
```
Comment (jotaylo) [inline make-options.json]: Ugh, didn't mean to add this change!

File: Tasks/DotNetCoreCLIV2/Tests/DotnetMockHelper.ts [hunk 2]
```diff
@@ -66,7 +65,7 @@ export class DotnetMockHelper {
     }
 
     public registerNugetUtilityMock(projectFile: string[]) {
-        this.tmr.registerMock('nuget-task-common/Utility', {
+        this.tmr.registerMock('packaging-common/nuget/Utility', {
             getPatternsArrayFromInput: function(input) {
                 return [`fromMockedUtility-${input}`];
             },
```
Comment (pspill) [inline Tasks/DotNetCoreCLIV2/Tests/DotnetMockHelper.ts]: getNuGetFeedRegistryUrl below doesn't match current parameter order.

File: Tasks/NuGetCommandV2/Tests/NugetMockHelper.ts [hunk 3]
```diff
@@ -67,7 +66,7 @@ export class NugetMockHelper {
     }
 
     public registerNugetUtilityMock(projectFile: string[]) {
-        this.tmr.registerMock('nuget-task-common/Utility', {
+        this.tmr.registerMock('packaging-common/nuget/Utility', {
             getPatternsArrayFromInput: function(input) {
                 return [input];
             },
```
Comment (pspill) [inline Tasks/NuGetCommandV2/Tests/NugetMockHelper.ts]: getNuGetFeedRegistryUrl below doesn't match current parameter order

File: Tasks/NuGetCommandV2/nugetrestore.ts [hunk 4]
```diff
@@ -2,18 +2,16 @@ import * as path from "path";
...
+import * as commandHelper from "packaging-common/nuget/CommandHelper";
+import {NuGetConfigHelper2} from "packaging-common/nuget/NuGetConfigHelper2";
+import * as ngToolRunner from "packaging-common/nuget/NuGetToolRunner2";
+import peParser = require("packaging-common/pe-parser/index");
+import {VersionInfo} from "packaging-common/pe-parser/VersionResource";
+import * as nutil from "packaging-common/nuget/Utility";
+import * as pkgLocationUtils from "packaging-common/locationUtilities";
 import * as telemetry from "utility-common/telemetry";
-import INuGetCommandOptions from "./Common/INuGetCommandOptions";
+import INuGetCommandOptions from "packaging-common/nuget/INuGetCommandOptions2";
```
Comment (pspill) [inline Tasks/NuGetCommandV2/nugetrestore.ts]: interesting... was this just an old mistake, this command was using the old tool runner environment settings?

Comment (jotaylo) [reply Tasks/NuGetCommandV2/nugetrestore.ts]: No, it was using the new ones, but it defined them in its own file for some reason.

File: Tasks/DotNetCoreCLIV2/Tests/DotnetMockHelper.ts (hunk 2 above)
Comment (jotaylo) [reply Tasks/DotNetCoreCLIV2/Tests/DotnetMockHelper.ts]: I plan to move these to a common location too, as soon as I get a chance :)

**Pull request merged**